### PR TITLE
Fix: python aws creds vpc ec2 classic

### PR DESCRIPTION
**Pull request by moliele**
Python file: Add AWS credential checks.
Terraform: add VPC
Docker: update images

### CODE REVIEW

File: core/docker/docker-compose.yaml [hunk 1]
```diff
@@ -59,12 +61,12 @@ services:
       - broker
       - schema-registry
     ports:
-      - 18083:18083
```
Comment (gianlucanatali) [inline core/docker/docker-compose.yaml:62]: why did you change the ports? I honestly don't remember why they were not standard... but just be sure it doesn't break smg else :) fine for me  @moliele

File: core/terraform/aws/main.tf [hunk 2]
```diff
@@ -1,12 +1,14 @@
...
-  secret_key = var.secret_key
+  #access_key = var.access_key #because gimme-aws-creds doesnt require this
+  #secret_key = var.secret_key #because gimme-aws-creds doesnt require this
+  profile    = var.profile
   default_tags {
       tags = {
           owner_email = var.owner_email
           purpose = var.purpose
           ref_link = var.ref_link
+          Deployed_By: "Terraform"
```
Comment (gianlucanatali) [inline core/terraform/aws/main.tf:11]: I like this @moliele

File: core/terraform/aws/variables.tf [hunk 3]
```diff
@@ -31,13 +31,13 @@ variable "ami" {
...
 }
 
-variable "access_key" {
-  description = "AWS Access Key"
-}
+#variable "access_key" {
+#  description = "AWS Access Key"
+#}
 
-variable "secret_key" {
```
Comment (gianlucanatali) [inline core/terraform/aws/variables.tf:38]: how is this not needed anymore? I have seen it in the props still. I think we need to keep it? @moliele

File: core/terraform/aws/workshop-core/main.tf [hunk 4]
```diff
@@ -241,7 +281,7 @@ resource "null_resource" "vm_provisioners" {
   //Adding AWS Credentials for Connect
   provisioner "file" {
     source      = "aws_credentials.txt"
-    destination = "~/.workshop/docker/.aws/credentials"
+    destination = ".workshop/docker/.aws/credentials"
```
Comment (gianlucanatali) [inline core/terraform/aws/workshop-core/main.tf:284]: have you tested extensions @moliele ? I think this is used by connectors. if the path now should be different fine for me, but let's make sure it doesn't break extensions connectors :) Like redshift extension as example, or S3

File: core/terraform/aws/workshop-core/variables.tf [hunk 5]
```diff
@@ -52,4 +52,20 @@ variable "onprem_topics" {
...
+  description = "VPC CIDR block"
+}
+
+variable "public_subnet_cidr_blocks" {
+  default     = ["10.0.101.0/24", "10.0.102.0/24"]
+  description = "List of public subnet CIDR blocks"
+}
+
+variable "availability_zones" {
+  default     = ["eu-west-2a", "eu-west-2b"]
```
Comment (gianlucanatali) [inline core/terraform/aws/workshop-core/variables.tf:68]: mmm... are these configurable? Can we use the region in the properties somehow? @moliele

File: workshop-example-aws.yaml [hunk 6]
```diff
@@ -15,9 +15,11 @@ workshop:
...
     # The workshop cloud provider & VM configuration
     cloud_provider: aws
-    access_key: <access key>
-    secret_key: <secret key>
-    region: eu-west-2
+    #access_key: <access key>
+    #secret_key: <secret key>
+    profile: <AWS profile>
+    region: <AWS preferred region>
+    availability_zones: <list of availability zones>
```
Comment (gianlucanatali) [inline workshop-example-aws.yaml:22]: so here you have the availability zones, so you could use those in the VPC I guess? @moliele

File: core/docker/docker-compose.yaml (hunk 1 above)
Comment (moliele) [reply core/docker/docker-compose.yaml:62]: Once I updated the images versions I faced with this error:
curl: (56) Recv failure: Connection reset by peer.

It was related to not using the standard ports. 

That's the reason why they have been changed, I changed them everywhere, but I will check again in order to avoid breaking something.

File: core/terraform/aws/variables.tf (hunk 3 above)
Comment (korzan-og) [reply core/terraform/aws/variables.tf:38]: If we use profile, these are not used automatically. So this is a case dependant on the previous comment.  https://github.com/confluentinc/confluent-hybrid-cloud-workshop/pull/63/files#r1070994255

File: core/terraform/aws/workshop-core/variables.tf (hunk 5 above)
Comment (korzan-og) [reply core/terraform/aws/workshop-core/variables.tf:68]: I think we should add these as variables to the workshop-example.yamls. I had a problem with overlapping with an existing subnet.

File: workshop-create.py [hunk 7]
```diff
@@ -6,89 +6,116 @@
...
+    try:
+        sts.get_caller_identity()
+        return True
+    except botocore.exceptions.UnauthorizedSSOTokenError:
+        return False
+    except botocore.exceptions.ClientError:
+        return False
+
+
+if (config['workshop']['core']['cloud_provider']) == 'aws':
```
Comment (korzan-og) [inline workshop-create.py:42]: Nice solution to the problem. Can we add likewise check to other clouds?

File: workshop-destroy.py [hunk 8]
```diff
@@ -4,21 +4,43 @@
...
+    try:
+        sts.get_caller_identity()
+        return True
+    except botocore.exceptions.UnauthorizedSSOTokenError:
+        return False
+    except botocore.exceptions.ClientError:
+        return False
+
+
+if (config['workshop']['core']['cloud_provider']) == 'aws':
```
Comment (korzan-og) [inline workshop-destroy.py:35]: Sames as above. Nice solution to the problem. Can we add likewise check to other clouds?

File: workshop-example-aws.yaml (hunk 6 above)
Comment (moliele) [reply workshop-example-aws.yaml:23]: @korzan-og Did you mean in this comment: https://github.com/confluentinc/confluent-hybrid-cloud-workshop/pull/63#discussion_r1072097897 , adding these lines? thanks!

Comment (korzan-og) [reply workshop-example-aws.yaml:23]: Yes. Thank you.

**Review by korzan-og — COMMENTED** [review]
Changes seem good.

File: workshop-create.py (hunk 7 above)
Comment (moliele) [reply workshop-create.py:42]: yes, working on that for gcp and azure

**Review by tjunderhill — APPROVED** [review]
LGTM!
